jaq-json/tests/funs.rs

```diff
@@ -77,3 +77,50 @@ fn tojson() {
     give(json!(0), "0.0 / 0.0 | tojson", json!("null"));
     give(json!(0), "1.0 / 0.0 | tojson", json!("null"));
 }
+
+#[test]
+fn math_rem() {
+    // generated with this command with modification for errors and float rounding
+    // cargo run -- -rn 'def f: -2, -1, 0, 2.1, 3, 4000000001; f as $a | f as $b | "give!(json!(null), \"\($a) / \($b)\", \(try ($a % $b) catch tojson));"'
+    // TODO: use fail!()?
+    give(json!(null), "-2 % -2", json!(0));
+    give(json!(null), "-2 % -1", json!(0));
+    give(json!(null), "try (-2 % 0) catch .", json!("cannot calculate -2 % 0"));
+    give(json!(null), "-2 % 2.1", json!(-2.0));
+    give(json!(null), "-2 % 3", json!(-2));
+    give(json!(null), "-2 % 4000000001", json!(-2));
+    give(json!(null), "-1 % -2", json!(-1));
+    give(json!(null), "-1 % -1", json!(0));
+    give(json!(null), "try (-1 % 0) catch .", json!("cannot calculate -1 % 0"));
+    give(json!(null), "-1 % 2.1", json!(-1.0));
+    give(json!(null), "-1 % 3", json!(-1));
+    give(json!(null), "-1 % 4000000001", json!(-1));
+    give(json!(null), "0 % -2", json!(0));
+    give(json!(null), "0 % -1", json!(0));
+    give(json!(null), "try (0 % 0) catch .", json!("cannot calculate 0 % 0"));
+    give(json!(null), "0 % 2.1", json!(0.0));
+    give(json!(null), "0 % 3", json!(0));
+    give(json!(null), "0 % 4000000001", json!(0));
+    give(json!(null), "2.1 % -2 | . * 1000 | round", json!(100));
+    give(json!(null), "2.1 % -1 | . * 1000 | round", json!(100));
+    give(json!(null), "2.1 % 0 | tojson", json!("null"));
```

Comment:
Yes. You could also use something like `give(json!(0), "1.0 / 0.0 | isinfinite", json!(true));` to check for the precise output of an arithmetic operation. (Use `isnan` where appropriate.)
The `tojson` test serves the particular purpose of checking `tojson`, so there, we lose information about NaN / Inf. But that's on purpose, because `tojson` *has* to lose that information.

Reply:
👍 in this case `2.1 % 0` ends up as NaN it seems. Test updated now.